services/mizu/monitor.go

```diff
@@ -37,10 +37,20 @@ func registerMetrics() {
 		memoryUsage := metrics.Memory.Usage
 		maxUsage := metrics.Memory.MaxUsage
 		memoryLimit := metrics.Memory.Limit
+		if memoryLimit == 0 {
+			// utils.Log("Container has stopped", utils.ErrorTAG)
+			// error needs to be handled in a better way
+			continue
```

Comment:
Even if the container has stopped we should store the metrics so that if these metrics are retrieved on the client-side we can show the client's application as offline. Also this will tell us for how much duration the container was offline.

Reply:
Yeah, so this was creating few other problems too:
* online CPUs and memory limit returned 0 so we had some metrics stored as NaN. I assumed this only happened when the container was offline.
* NaN in metrics posed some problems in the final JSON parsing.